source/extensions/transport_sockets/tls/cert_validator/cert_validator.h

```diff
@@ -46,7 +46,10 @@ struct ValidationResults {
 class CertValidator {
 public:
   // Wraps cert validation parameters added from time to time.
-  struct ExtraValidationContext {};
+  struct ExtraValidationContext {
+    // Optional reference to local address instance.
+    Envoy::OptRef<const Envoy::Network::Address::Instance> local_address;
```

Comment:
Address::InstanceConstSharedPtr is a better choice here, since it is already a shared pointer. null would indicate it was not set. Posting a reference into an asynchronous routing is brittle as you need to be sure that lifetime of referent is sufficient for completion of cert validation (and all error cases).